### PR TITLE
MGMT-14226: Upgrade moby to 20.10.24

### PR DESCRIPTION
**Pull request by CrystalChun**
[MGMT-14226](https://issues.redhat.com/browse/MGMT-14226)
Upgrades go mod github.com/moby/moby to version
20.10.24 for CVEs in the issue.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: vendor/github.com/moby/moby/pkg/ioutils/fswriters.go [hunk 1]
```diff
@@ -11,7 +10,7 @@ import (
 // temporary file and closing it atomically changes the temporary file to
 // destination path. Writing and closing concurrently is not allowed.
 func NewAtomicFileWriter(filename string, perm os.FileMode) (io.WriteCloser, error) {
-	f, err := ioutil.TempFile(filepath.Dir(filename), ".tmp-"+filepath.Base(filename))
+	f, err := os.CreateTemp(filepath.Dir(filename), ".tmp-"+filepath.Base(filename))
```
Comment (paul-maidment) [inline vendor/github.com/moby/moby/pkg/ioutils/fswriters.go:13]: Can some explanation be provided for why this change is needed.

I see that the approach in the current version of moby is also to use `ioutil.Tempfile`, my question is, why does this need to change? Especially as a modification to a vendor package?

File: vendor/github.com/moby/moby/pkg/ioutils/fswriters.go [hunk 2]
```diff
@@ -94,7 +93,7 @@ type AtomicWriteSet struct {
 // commit. If no temporary directory is given the system
 // default is used.
 func NewAtomicWriteSet(tmpDir string) (*AtomicWriteSet, error) {
-	td, err := ioutil.TempDir(tmpDir, "write-set-")
+	td, err := os.MkdirTemp(tmpDir, "write-set-")
```
Comment (paul-maidment) [inline vendor/github.com/moby/moby/pkg/ioutils/fswriters.go:96]: Same question here, why does this need to change? What is the motivation?

File: vendor/github.com/moby/moby/pkg/ioutils/temp_unix.go [hunk 3]
```diff
@@ -3,9 +3,9 @@
...
 package ioutils // import "github.com/docker/docker/pkg/ioutils"
 
-import "io/ioutil"
+import "os"
 
-// TempDir on Unix systems is equivalent to ioutil.TempDir.
+// TempDir on Unix systems is equivalent to os.MkdirTemp.
 func TempDir(dir, prefix string) (string, error) {
-	return ioutil.TempDir(dir, prefix)
+	return os.MkdirTemp(dir, prefix)
```
Comment (paul-maidment) [inline vendor/github.com/moby/moby/pkg/ioutils/temp_unix.go:10]: Same question here

File: vendor/github.com/moby/moby/pkg/ioutils/temp_windows.go [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+	"os"
 
 	"github.com/docker/docker/pkg/longpath"
 )
 
-// TempDir is the equivalent of ioutil.TempDir, except that the result is in Windows longpath format.
+// TempDir is the equivalent of os.MkdirTemp, except that the result is in Windows longpath format.
 func TempDir(dir, prefix string) (string, error) {
-	tempDir, err := ioutil.TempDir(dir, prefix)
+	tempDir, err := os.MkdirTemp(dir, prefix)
```
Comment (paul-maidment) [inline vendor/github.com/moby/moby/pkg/ioutils/temp_windows.go:11]: Same question here

File: vendor/github.com/moby/moby/pkg/ioutils/fswriters.go (hunk 1 above)
Comment (paul-maidment) [reply vendor/github.com/moby/moby/pkg/ioutils/fswriters.go:13]: Also, just looked at the original motivation (the CVE) and this does not appear to be a CVE about file management or temporary file issues, so this change seems unrelated?

**Review by carbonin — APPROVED** [review]
